src/ai/defaultai.cc

```diff
@@ -3330,18 +3330,30 @@ void DefaultAI::check_flag_distances(const Time& gametime) {
 // Dealing with diplomacy actions
 void DefaultAI::diplomacy_actions(const Time& gametime) {
 
+	// don't do any diplomacy for the first 10 + x minutes to avoid click races for allies
+	if (gametime < Time((10 + RNG::static_rand(10)) * 60 * 1000)) {
+		return;
+	}
+
+	const Widelands::PlayerNumber mypn = player_number();
+	const Widelands::Player* me = game().get_player(mypn);
+
+	// if we are defeated or the last one in a team leave team
+	if (me->team_number() != 0 &&
+	    (player_statistics.members_in_team(me->team_number()) == 1 || me->is_defeated())) {
+		game().send_player_diplomacy(mypn, Widelands::DiplomacyAction::kLeaveTeam, 0 /* ignored */);
+		verb_log_dbg_time(gametime, "Player(%d), leaves team (%d) as last one.\n",
```

Comment:
Please include something to make it easy to filter the log in one pass. E.g.

```suggestion
		verb_log_dbg_time(gametime, "AI diplomacy: Player(%d), leaves team (%d) as last one.\n",
```

Reply:
done